src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/DefaultBindingMetadataProvider.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Metadata
+{
+    /// <summary>
+    /// A default implementation of <see cref="IBindingMetadataProvider"/>.
+    /// </summary>
+    public class DefaultBindingMetadataProvider : IBindingMetadataProvider
+    {
+        /// <inheritdoc />
+        public void GetBindingMetadata([NotNull] BindingMetadataProviderContext context)
+        {
+            // For Model Name  - we only use the first attribute we find. An attribute on the parameter
```

Comment:
The more I think about it, the more it makes sense to keep this consistent with rest of the stuff. 
For model name specifically, I find it hard to imagine an attribute on the type defining a model name, having said that should we instead fix the BindAttribute to treat null values as empty?